Con 2367 input fields validation bug p2sprint10

karpagamNallasamy opened the pull request:
Bug #3261,#3256,3288,#3270,#3262. buddy testing done by Nishadi

karpagamNallasamy committed:
3256 Giving an error when adding users and roles to a group and editing a group

karpagamNallasamy committed:
#3261 Group Name can be created with all allowed special characters

karpagamNallasamy committed:
3288 User can to create contacts with * on contact point name using APIs

VijayHirudayasamy-bc commented on api/CcsSso.Shared/Domain/Helpers/UtilityHelper.cs:
```diff
@@ -74,12 +74,20 @@ public static bool IsPasswordValidForRequiredCharactors(string password)
+      bool IsRestrictedChar = false;
       try
       {
+        string[] tokens = "+ * % # ! $".Split();
```
Can you use normal array with list of item in it string[] cars = {"Volvo", "BMW", "Ford", "Mazda"};

VijayHirudayasamy-bc commented on api/CcsSso.Shared/Domain/Helpers/UtilityHelper.cs:
```diff
@@ -107,11 +115,21 @@ public static bool IsUserNameLengthValid(string Name)
-        Regex regex = new Regex(RegexExpression.VALID_GROUP_NAME);
-        Match match = regex.Match(Name);
-        return match.Success;
+        string[] tokens = "* % ! $".Split();
```
same here

VijayHirudayasamy-bc commented on api/CcsSso.Shared/Domain/Helpers/UtilityHelper.cs:
```diff
@@ -121,11 +139,21 @@ public static bool IsGroupNameValid(string Name)
-        Regex regex = new Regex(RegexExpression.VALID_CONTACT_NAME);
-        Match match = regex.Match(Name);
-        return match.Success;
+        string[] tokens = "+ * % # ! $".Split();
```
here too

VijayHirudayasamy-bc commented on api/CcsSso.Core.Service/External/OrganisationGroupService.cs:
```diff
@@ -187,11 +195,22 @@ public async Task UpdateGroupAsync(string ciiOrganisationId, int groupId, Organi
       }
 
+      if (!string.IsNullOrWhiteSpace(organisationGroupRequestInfo.GroupName))
+      {
```
Can be keep all the validation as a one function

VijayHirudayasamy-bc reviewed (commented):
Please check the comments

karpagamNallasamy committed:
restricted special charactor array with list of item

karpagamNallasamy commented:
Can you use normal array with list of item in it string[] cars = {"Volvo", "BMW", "Ford", "Mazda"} – done

each validation different API call. make common function not good. also it takes time now.

I am trying many time to reach you by teams call.



It is very urgent. Nishadi,Tina asking me consciously. please merge it.


Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

________________________________
From: VijayHirudayasamy-bc ***@***.***>
Sent: Thursday, June 30, 2022 4:20:22 PM
To: Crown-Commercial-Service/conclave-ssso ***@***.***>
Cc: Karpagam Nallasamy ***@***.***>; Author ***@***.***>
Subject: Re: [Crown-Commercial-Service/conclave-ssso] Con 2367 input fields validation bug p2sprint10 (PR #697)


@VijayHirudayasamy-bc commented on this pull request.

Please check the comments

________________________________

In api/CcsSso.Shared/Domain/Helpers/UtilityHelper.cs<https://github.com/Crown-Commercial-Service/conclave-ssso/pull/697#discussion_r910879690>:

>        try
       {
+        string[] tokens = "+ * % # ! $".Split();


Can you use normal array with list of item in it string[] cars = {"Volvo", "BMW", "Ford", "Mazda"};

________________________________

In api/CcsSso.Shared/Domain/Helpers/UtilityHelper.cs<https://github.com/Crown-Commercial-Service/conclave-ssso/pull/697#discussion_r910879996>:

>        try
       {
-        Regex regex = new Regex(RegexExpression.VALID_GROUP_NAME);
-        Match match = regex.Match(Name);
-        return match.Success;
+        string[] tokens = "* % ! $".Split();


same here

________________________________

In api/CcsSso.Shared/Domain/Helpers/UtilityHelper.cs<https://github.com/Crown-Commercial-Service/conclave-ssso/pull/697#discussion_r910880123>:

>        try
       {
-        Regex regex = new Regex(RegexExpression.VALID_CONTACT_NAME);
-        Match match = regex.Match(Name);
-        return match.Success;
+        string[] tokens = "+ * % # ! $".Split();


here too

________________________________

In api/CcsSso.Core.Service/External/OrganisationGroupService.cs<https://github.com/Crown-Commercial-Service/conclave-ssso/pull/697#discussion_r910881825>:

>        {
         throw new CcsSsoException(ErrorConstant.ErrorInvalidGroupName);
       }

+      if (!string.IsNullOrWhiteSpace(organisationGroupRequestInfo.GroupName))
+      {


Can be keep all the validation as a one function

—
Reply to this email directly, view it on GitHub<https://github.com/Crown-Commercial-Service/conclave-ssso/pull/697#pullrequestreview-1024657700>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AXH6RBL3JIEFJMZLWY6XIBLVRV3W5ANCNFSM52IH5ENA>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

------------------------------------------------------------------------------------------------------------ This transmission has been sent by either Brickendon Consulting Limited, Brickendon Consulting (US) Inc. or Brickendon Consulting (Poland) Sp. z o.o., referred to below as “Brickendon”. This email and any attachments relating to the official business of Brickendon is proprietary to Brickendon, confidential and may be privileged. This message is intended solely for the addressee(s) of this email who are the sole authorised recipient(s). If you have received this message in error, please send it back to us, and immediately and permanently delete it. Do not use, copy, or disclose the information contained in this message or in any attachment or base any action on it. Brickendon cannot ensure that the integrity of this email has been maintained nor that it is free of errors, virus, interception or interference. The sender therefore does not accept liability for any errors or omissions in the contents of this message which arise as a result of e-mail transmission. You should check that any attachments are virus free before opening them. Any views or statements in attachments are (unless the context or letter sign off otherwise makes it clear) solely those of the author and do not necessarily represent those of Brickendon. Brickendon Consulting Limited is registered in England with company number 07262520 and whose registered office is at 4 Prince Albert Road, London, NW1 7SN. Brickendon Consulting (US) Inc. is registered in the United States of America with company number 6022535 and whose registered office is at 3500 South Dupont Highway, Dover DE 19901. Brickendon Consulting (Poland) Sp. z o.o. is registered in Poland with company number KRS000070318 and whose registered office is at Ulica Przemyslawa Gintrowskiego 31, 02-697 Warzawa. More information on Brickendon is available at www.brickendon.com -----------------------------------------------------------------------------------------------------------------------------
